tests/TestCase/Http/ClientTest.php

```diff
@@ -898,4 +920,98 @@ public function testRedirectDifferentSubDomains()
         $this->assertInstanceOf(Response::class, $result);
         $this->assertSame($response, $result);
     }
+
+    /**
+     * Scheme is set when passed to client in string
+     */
+    public function testCreateFromUrlSetsScheme()
+    {
+        $client = Client::createFromUrl('https://example.co/');
+        $this->assertSame('https', $client->getConfig('scheme'));
+    }
+
+    /**
+     * Host is set when passed to client in string
+     */
+    public function testCreateFromUrlSetsHost()
+    {
+        $client = Client::createFromUrl('https://example.co/');
+        $this->assertSame('example.co', $client->getConfig('host'));
+    }
+
+    /**
+     * basePath is set when passed to client in string
+     */
+    public function testCreateFromUrlSetsBasePath()
```

Comment:
Instead of individual test methods like these you can add a provider method for URLs and then a single test methods which asserts the scheme, host, port and basePath generated.